streams/src/test/java/org/apache/kafka/streams/integration/QueryableStateIntegrationTest.java

```diff
@@ -509,6 +509,94 @@ public void queryOnRebalance() throws Exception {
         }
     }
 
+    @Test
+    public void shouldBeAbleQueryStandbyStateDuringRebalance() throws Exception {
```

Comment:
tthese `verifyAllKVKeys` methods will call ```  final ReadOnlyKeyValueStore<String, Long> store =
                        streamsWithKey.store(storeName, QueryableStoreTypes.keyValueStore());```, which should fail by default on the standby? (given thats changed from original PR)?

Reply:
Worth having two tests for both modes?

Reply:
yeah, that's correct. The test does fail on this version of the code. Nice catch!

The test for active replicas would pass regardless, it's just a question of whether it returns the data during recovery or after the transition to running. But the test you added for standbys would only pass if we allow querying stale state.

I'm not sure it makes sense to add a negative test for standbys, though, since there are already tests ensuring that we throw the appropriate exceptions when using the old method (with a default of no stale stores). WDYT?

Reply:
makes sense...  I will take another look at this when I rebase against the lag PR